### PR TITLE
chore: adds codecov

### PR DESCRIPTION
**Pull request by rluders**
This PR adds the codecov to the repository.

## Categories

- [ ] Bugfix
- [ ] Data Enhancement
- [ ] Feature
- [ ] Backporting
- [X] Others (CI, Infrastructure, Documentation)

## Sample Archive

N/A

## Documentation

N/A

## Unit Tests

N/A

## Privacy

N/A

## Breaking Changes

No

## References

https://issues.redhat.com/browse/CCXDEV-10925

### CODE REVIEW

**Review by ncaak — APPROVED** [review]

File: .codecov.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    require_ci_to_pass: no
+
+coverage:
+  precision: 2
+  round: down
+  range: "20...100"
+  status:
+    project: no
+    patch: false
+    changes: false
```
Comment (tremes) [inline .codecov.yml:12]: Just out of curiosity. Is there some docs for these attributes?

Comment (rluders) [reply .codecov.yml:12]: Well, the Codecov documentation should have it.

Comment (rluders) [reply .codecov.yml:12]: https://docs.codecov.com/docs/codecov-yaml

File: .codecov.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+codecov:
+  notify:
+    require_ci_to_pass: no
+
+coverage:
+  precision: 2
+  round: down
+  range: "20...100"
```
Comment (tremes) [inline .codecov.yml:8]: I think the second number set to 100 doesn't make much sense when reading https://docs.codecov.com/docs/coverage-configuration#range. I would go with 70 or 75

File: .codecov.yml (hunk 1 above)
Comment (tremes) [reply .codecov.yml:12]: Are `project` and `patch` attributes set correctly? I was looking at https://docs.codecov.com/docs/commit-status#disabling-a-status

File: .codecov.yml (hunk 2 above)
Comment (rluders) [reply .codecov.yml:8]: @tremes check if DVO doesn’t have the same config. If I’m not wrong I copied the same config, maybe we want to change it there as well.

File: .codecov.yml (hunk 1 above)
Comment (rluders) [reply .codecov.yml:12]: AFAIK, both should works but no problem to change it to `off`.

Comment (rluders) [reply .codecov.yml:12]:
```sh
➜ curl --data-binary @.codecov.yml https://codecov.io/validate

Valid!

{
  "codecov": {
    "notify": {},
    "require_ci_to_pass": false
  },
  "coverage": {
    "precision": 2,
    "round": "down",
    "range": [
      20.0,
      100.0
    ],
    "status": {
      "project": false,
      "patch": false,
      "changes": false
    }
  },
  "parsers": {
    "gcov": {
      "branch_detection": {
        "conditional": true,
        "loop": true,
        "method": false,
        "macro": false
      }
    }
  },
  "comment": {
    "layout": "reach,diff,flags,tree",
    "behavior": "default",
    "require_changes": false
  }
}
```

Never mind I'll not change it... I checked and the validator seems to be fine with it.

Comment (tremes) [reply .codecov.yml:12]: OK thanks and no problem, but now the range `range: "70...75"` seems to strict to me. :) I would go something like `40..75` or `30..75`

Comment (rluders) [reply .codecov.yml:12]: OK. Let me change it to 30...75. :)

**Review by ncaak — APPROVED** [review]
/retest

**Comment by rluders** [comment]
/override lint

Lint is failing for other reasons, not 'cause of the changes on this PR.

**Comment by rluders** [comment]
/override ci/prow/lint

Lint is failing for other reasons, not 'cause of the changes on this PR.

**Pull request merged**